product_deletion_utility/components/delete.py

```diff
@@ -0,0 +1,711 @@
+            secret_namespace (str): The namespace of the secret.
+        Returns:
+            None. Updates os.environ as is expected by Nexusctl.
+        """
+        try:
+            secret = self.k8s_client.read_namespaced_secret(
+                secret_name, secret_namespace
+            )
+        except (MaxRetryError, ApiException):
+            print(f'WARNING: unable to read Kubernetes secret {secret_namespace}/{secret_name}')
+            return
+        if secret.data is None:
+            print(f'WARNING: unable to read Kubernetes secret {secret_namespace}/{secret_name}')
+            return
+
+        os.environ.update({
+            'NEXUS_USERNAME': b64decode(secret.data['username']).decode(),
+            'NEXUS_PASSWORD': b64decode(secret.data['password']).decode()
+        })
+
+    def __init__(self, catalogname=PRODUCT_CATALOG_CONFIG_MAP_NAME,
+                 catalognamespace=PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+                 productname=None,
+                 productversion=None,
+                 nexus_url=DEFAULT_NEXUS_URL,
+                 docker_url=DEFAULT_DOCKER_URL,
+                 nexus_credentials_secret_name=NEXUS_CREDENTIALS_SECRET_NAME,
+                 nexus_credentials_secret_namespace=NEXUS_CREDENTIALS_SECRET_NAMESPACE):
+
+        self.pname = productname
+        self.pversion = productversion
+        self.catalogname = catalogname
+        self.catalognamespace = catalognamespace
+        self.uninstall_component = UninstallComponents()
+        self.k8s_client = self._get_k8s_api()
+        self._update_environment_with_nexus_credentials(
+            nexus_credentials_secret_name, nexus_credentials_secret_namespace
+        )
+        self.docker_api = DockerApi(DockerClient(docker_url))
+        self.nexus_api = NexusApi(NexusClient(nexus_url))
+        print(f'catalog name and namespace are {self.catalogname}, {self.catalognamespace}')
+        # inheriting the properties of parent ProductCatalog class
+        super().__init__(self.catalogname, self.catalognamespace)
+
+
+    def remove_product_docker_images(self):
+        """Remove a product's Docker images.
+        This function will only remove images that are not used by another
+        product in the catalog. For images that are used by another
+        Args:
+            None
+        Returns:
+            None
+        Raises:
+            ProductInstallException: If an error occurred removing an image.
+        """
+        product = self.get_product(self.pname, self.pversion)
+        images_to_remove = product.docker_images
+        print(f'Images to remove are - {images_to_remove}')        
+        if not images_to_remove:
+            print(f"No docker images found in the configmap data for {self.pname}:{self.pversion}")
+            return
+        other_products = [
+            p for p in self.products
+            if p.version != product.version or p.name != product.name
```

Comment:
This is picking on a small detail, but I think it is easier to understand if you reverse the order.

if p.name != product.name or p.version != product.version 

If the name's do not match, obviously, the products do not match.  But, if the names do match, but their versions do not match, then they _also_ are not a match.

Reply:
I added a commit that rearranged the if statement at each location that I commented. I figured if I was going to complain about it, I could help fix it.

Reply:
:) 
Sure. Your commit works. Thanks